File: crewai/config/market.py
"""Market Configuration for PriceWaze CrewAI system.

Centralizes all market-specific settings for multi-region support.
This mirrors the TypeScript market configuration in src/config/market.ts
"""

import os
from dataclasses import dataclass
from functools import lru_cache
from typing import Literal


MarketCode = Literal["DO", "US", "MX", "ES", "CO", "global"]


@dataclass(frozen=True)
class CurrencyConfig:
    """Currency configuration for a market."""
    code: str
    symbol: str
    locale: str


@dataclass(frozen=True)
class MapConfig:
    """Map configuration for a market."""
    center: tuple[float, float]  # (lng, lat)
    zoom: int


@dataclass(frozen=True)
class LegalConfig:
    """Legal configuration for a market."""
    jurisdiction: str
    disclaimer_es: str
    disclaimer_en: str
    contract_law: str


@dataclass(frozen=True)
class AIConfig:
    """AI configuration for a market."""
    market_context: str
    price_unit: str
    key_locations: list[str]  # Major cities/regions for backstory


@dataclass(frozen=True)
class MarketConfig:
    """Complete market configuration."""
    code: MarketCode
    name: str
    currency: CurrencyConfig
    map: MapConfig
    legal: LegalConfig
    ai: AIConfig


# Market configurations matching TypeScript definitions
MARKET_CONFIGS: dict[MarketCode, MarketConfig] = {
    "DO": MarketConfig(
        code="DO",
        name="Dominican Republic",
        currency=CurrencyConfig(code="DOP", symbol="RD$", locale="es-DO"),
        map=MapConfig(center=(-69.9312, 18.4861), zoom=12),
        legal=LegalConfig(
            jurisdiction="Dominican Republic",
            disclaimer_es="Consulte con un abogado licenciado en la República Dominicana",
            disclaimer_en="Consult with a licensed attorney in the Dominican Republic",
            contract_law="Dominican Republic Civil Code and Property Law",
        ),
        ai=AIConfig(
            market_context="Dominican Republic real estate market",
            price_unit="DOP (Dominican Pesos)",
            key_locations=["Santo Domingo", "Punta Cana", "Santiago", "Puerto Plata"],
        ),
    ),
    "US": MarketConfig(
        code="US",
        name="United States",
        currency=CurrencyConfig(code="USD", symbol="$", locale="en-US"),
        map=MapConfig(center=(-98.5795, 39.8283), zoom=4),
        legal=LegalConfig(
            jurisdiction="United States",
            disclaimer_es="Consulte con un abogado licenciado en su estado",
            disclaimer_en="Consult with a licensed attorney in your state",
            contract_law="State real estate laws apply",
        ),
        ai=AIConfig(
            market_context="United States real estate market",
            price_unit="USD (US Dollars)",
            key_locations=["New York", "Los Angeles", "Miami", "Chicago"],
        ),
    ),
    "MX": MarketConfig(
        code="MX",
        name="Mexico",
        currency=CurrencyConfig(code="MXN", symbol="MX$", locale="es-MX"),
        map=MapConfig(center=(-99.1332, 19.4326), zoom=10),
        legal=LegalConfig(
            jurisdiction="Mexico",
            disclaimer_es="Consulte con un abogado licenciado en México",
            disclaimer_en="Consult with a licensed attorney in Mexico",
            contract_law="Mexican Civil Code and Property Law",
        ),
        ai=AIConfig(
            market_context="Mexican real estate market",
            price_unit="MXN (Mexican Pesos)",
            key_locations=["Mexico City", "Guadalajara", "Monterrey", "Cancún"],
        ),
    ),
    "ES": MarketConfig(
        code="ES",
        name="Spain",
        currency=CurrencyConfig(code="EUR", symbol="€", locale="es-ES"),
        map=MapConfig(center=(-3.7038, 40.4168), zoom=10),
        legal=LegalConfig(
            jurisdiction="Spain",
            disclaimer_es="Consulte con un abogado licenciado en España",
            disclaimer_en="Consult with a licensed attorney in Spain",
            contract_law="Spanish Civil Code and Property Law",
        ),
        ai=AIConfig(
            market_context="Spanish real estate market",
            price_unit="EUR (Euros)",
            key_locations=["Madrid", "Barcelona", "Valencia", "Sevilla"],
        ),
    ),
    "CO": MarketConfig(
        code="CO",
        name="Colombia",
        currency=CurrencyConfig(code="COP", symbol="COP$", locale="es-CO"),
        map=MapConfig(center=(-74.0721, 4.7110), zoom=10),
        legal=LegalConfig(
            jurisdiction="Colombia",
            disclaimer_es="Consulte con un abogado licenciado en Colombia",
            disclaimer_en="Consult with a licensed attorney in Colombia",
            contract_law="Colombian Civil Code and Property Law",
        ),
        ai=AIConfig(
            market_context="Colombian real estate market",
            price_unit="COP (Colombian Pesos)",
            key_locations=["Bogotá", "Medellín", "Cartagena", "Cali"],
        ),
    ),
    "global": MarketConfig(
        code="global",
        name="Global",
        currency=CurrencyConfig(code="USD", symbol="$", locale="en-US"),
        map=MapConfig(center=(0, 20), zoom=2),
        legal=LegalConfig(
            jurisdiction="varies by location",
            disclaimer_es="Consulte con un abogado licenciado en su jurisdicción local",
            disclaimer_en="Consult with a licensed attorney in your local jurisdiction",
            contract_law="Local property laws apply",
        ),
        ai=AIConfig(
            market_context="international real estate market",
            price_unit="USD (US Dollars)",
            key_locations=["major metropolitan areas"],
        ),
    ),
}


@lru_cache
def get_market_config() -> MarketConfig:
    """
    Get the current market configuration.

    Reads from NEXT_PUBLIC_MARKET_CODE env var, defaults to 'global'.

    Returns:
        MarketConfig for the current market
    """
    market_code = os.environ.get("NEXT_PUBLIC_MARKET_CODE", "global")
    if market_code not in MARKET_CONFIGS:
        market_code = "global"
    return MARKET_CONFIGS[market_code]  # type: ignore


def get_market_config_by_code(code: MarketCode) -> MarketConfig:
    """
    Get a specific market configuration by code.

    Args:
        code: Market code (DO, US, MX, ES, CO, global)

    Returns:
        MarketConfig for the specified market
    """
    return MARKET_CONFIGS.get(code, MARKET_CONFIGS["global"])


def format_price(amount: float, market: MarketConfig | None = None) -> str:
    """
    Format price according to market locale.

    Args:
        amount: Price amount
        market: Optional market config (uses current market if not provided)

    Returns:
        Formatted price string
    """
    config = market or get_market_config()
    # Simple formatting - for more complex locale support, use babel
    formatted = f"{amount:,.0f}"
    return f"{config.currency.symbol}{formatted}"


def get_ai_market_context() -> str:
    """
    Get AI prompt context for the current market.

    Returns:
        Context string for AI prompts
    """
    config = get_market_config()
    return (
        f"You are a real estate analyst for the {config.ai.market_context}. "
        f"Prices are typically quoted in {config.ai.price_unit}."
    )


def get_agent_backstory_locations() -> str:
    """
    Get key locations string for agent backstories.

    Returns:
        Comma-separated list of key market locations
    """
    config = get_market_config()
    return ", ".join(config.ai.key_locations)

File: src/config/market.ts
/**
 * Market Configuration System
 * Centralizes all market-specific settings for multi-region support
 */

export type MarketCode = 'DO' | 'US' | 'MX' | 'ES' | 'CO' | 'global';

export interface MarketConfig {
  code: MarketCode;
  name: string;
  currency: {
    code: string;
    symbol: string;
    locale: string;
  };
  map: {
    center: [number, number]; // [lng, lat]
    zoom: number;
    bounds?: [[number, number], [number, number]];
  };
  legal: {
    jurisdiction: string;
    disclaimerEs: string;
    disclaimerEn: string;
    contractLaw: string;
  };
  ai: {
    marketContext: string;
    priceUnit: string;
  };
  seo: {
    titleSuffix: string;
    description: string;
    keywords: string[];
  };
}

const MARKET_CONFIGS: Record<MarketCode, MarketConfig> = {
  DO: {
    code: 'DO',
    name: 'Dominican Republic',
    currency: {
      code: 'DOP',
      symbol: 'RD$',
      locale: 'es-DO',
    },
    map: {
      center: [-69.9312, 18.4861], // Santo Domingo
      zoom: 12,
      bounds: [[-72.0, 17.5], [-68.3, 20.0]],
    },
    legal: {
      jurisdiction: 'Dominican Republic',
      disclaimerEs: 'Consulte con un abogado licenciado en la República Dominicana',
      disclaimerEn: 'Consult with a licensed attorney in the Dominican Republic',
      contractLaw: 'Dominican Republic Civil Code and Property Law',
    },
    ai: {
      marketContext: 'Dominican Republic real estate market',
      priceUnit: 'DOP (Dominican Pesos)',
    },
    seo: {
      titleSuffix: 'Dominican Republic',
      description: 'Real estate intelligence for the Dominican Republic market',
      keywords: ['Dominican Republic', 'Santo Domingo', 'Punta Cana', 'real estate'],
    },
  },
  US: {
    code: 'US',
    name: 'United States',
    currency: {
      code: 'USD',
      symbol: '$',
      locale: 'en-US',
    },
    map: {
      center: [-98.5795, 39.8283], // Geographic center of US
      zoom: 4,
    },
    legal: {
      jurisdiction: 'United States',
      disclaimerEs: 'Consulte con un abogado licenciado en su estado',
      disclaimerEn: 'Consult with a licensed attorney in your state',
      contractLaw: 'State real estate laws apply',
    },
    ai: {
      marketContext: 'United States real estate market',
      priceUnit: 'USD (US Dollars)',
    },
    seo: {
      titleSuffix: 'United States',
      description: 'AI-powered real estate intelligence for the US market',
      keywords: ['USA', 'real estate', 'property', 'homes'],
    },
  },
  MX: {
    code: 'MX',
    name: 'Mexico',
    currency: {
      code: 'MXN',
      symbol: 'MX$',
      locale: 'es-MX',
    },
    map: {
      center: [-99.1332, 19.4326], // Mexico City
      zoom: 10,
    },
    legal: {
      jurisdiction: 'Mexico',
      disclaimerEs: 'Consulte con un abogado licenciado en México',
      disclaimerEn: 'Consult with a licensed attorney in Mexico',
      contractLaw: 'Mexican Civil Code and Property Law',
    },
    ai: {
      marketContext: 'Mexican real estate market',
      priceUnit: 'MXN (Mexican Pesos)',
    },
    seo: {
      titleSuffix: 'Mexico',
      description: 'Inteligencia inmobiliaria para el mercado mexicano',
      keywords: ['México', 'bienes raíces', 'propiedades', 'inmuebles'],
    },
  },
  ES: {
    code: 'ES',
    name: 'Spain',
    currency: {
      code: 'EUR',
      symbol: '€',
      locale: 'es-ES',
    },
    map: {
      center: [-3.7038, 40.4168], // Madrid
      zoom: 10,
    },
    legal: {
      jurisdiction: 'Spain',
      disclaimerEs: 'Consulte con un abogado licenciado en España',
      disclaimerEn: 'Consult with a licensed attorney in Spain',
      contractLaw: 'Spanish Civil Code and Property Law',
    },
    ai: {
      marketContext: 'Spanish real estate market',
      priceUnit: 'EUR (Euros)',
    },
    seo: {
      titleSuffix: 'Spain',
      description: 'Inteligencia inmobiliaria para el mercado español',
      keywords: ['España', 'inmobiliaria', 'propiedades', 'viviendas'],
    },
  },
  CO: {
    code: 'CO',
    name: 'Colombia',
    currency: {
      code: 'COP',
      symbol: 'COP$',
      locale: 'es-CO',
    },
    map: {
      center: [-74.0721, 4.7110], // Bogotá
      zoom: 10,
    },
    legal: {
      jurisdiction: 'Colombia',
      disclaimerEs: 'Consulte con un abogado licenciado en Colombia',
      disclaimerEn: 'Consult with a licensed attorney in Colombia',
      contractLaw: 'Colombian Civil Code and Property Law',
    },
    ai: {
      marketContext: 'Colombian real estate market',
      priceUnit: 'COP (Colombian Pesos)',
    },
    seo: {
      titleSuffix: 'Colombia',
      description: 'Inteligencia inmobiliaria para el mercado colombiano',
      keywords: ['Colombia', 'Bogotá', 'Medellín', 'finca raíz'],
    },
  },
  global: {
    code: 'global',
    name: 'Global',
    currency: {
      code: 'USD',
      symbol: '$',
      locale: 'en-US',
    },
    map: {
      center: [0, 20],
      zoom: 2,
    },
    legal: {
      jurisdiction: 'varies by location',
      disclaimerEs: 'Consulte con un abogado licenciado en su jurisdicción local',
      disclaimerEn: 'Consult with a licensed attorney in your local jurisdiction',
      contractLaw: 'Local property laws apply',
    },
    ai: {
      marketContext: 'international real estate market',
      priceUnit: 'USD (US Dollars)',
    },
    seo: {
      titleSuffix: '',
      description: 'AI-powered real estate intelligence platform',
      keywords: ['real estate', 'property', 'pricing', 'negotiation'],
    },
  },
};

/**
 * Get the current market configuration
 * Reads from NEXT_PUBLIC_MARKET_CODE env var, defaults to 'global'
 */
export function getMarketConfig(): MarketConfig {
  const marketCode = (process.env.NEXT_PUBLIC_MARKET_CODE as MarketCode) || 'global';
  return MARKET_CONFIGS[marketCode] || MARKET_CONFIGS.global;
}

/**
 * Get a specific market configuration by code
 */
export function getMarketConfigByCode(code: MarketCode): MarketConfig {
  return MARKET_CONFIGS[code] || MARKET_CONFIGS.global;
}

/**
 * Get all available market configurations
 */
export function getAllMarkets(): MarketConfig[] {
  return Object.values(MARKET_CONFIGS);
}

/**
 * Format price according to market locale
 */
export function formatPrice(amount: number, market?: MarketConfig): string {
  const config = market || getMarketConfig();
  return new Intl.NumberFormat(config.currency.locale, {
    style: 'currency',
    currency: config.currency.code,
    minimumFractionDigits: 0,
    maximumFractionDigits: 0,
  }).format(amount);
}

/**
 * Get AI prompt context for the current market
 */
export function getAIMarketContext(): string {
  const config = getMarketConfig();
  return `You are a real estate analyst for the ${config.ai.marketContext}. Prices are typically quoted in ${config.ai.priceUnit}.`;
}

/**
 * Get legal disclaimer in specified language
 */
export function getLegalDisclaimer(lang: 'es' | 'en' = 'en'): string {
  const config = getMarketConfig();
  return lang === 'es' ? config.legal.disclaimerEs : config.legal.disclaimerEn;
}

// Export current market config for easy access
export const currentMarket = getMarketConfig();
